libbeat/processors/actions/replace.go

```diff
@@ -42,6 +42,21 @@ type replaceStringConfig struct {
 	FailOnError   bool            `config:"fail_on_error"`
 }
 
+func (strCfg replaceStringConfig) validate() error {
```

Comment:
This looks like we just need the presence of these fields.  Any reason not to change the stuct tags and let Unpack handle it?

```
type replaceConfig struct {
	Field       string         `config:"field" validate:"required"`
	Pattern     *regexp.Regexp `config:"pattern" validate:"required"`
	Replacement string         `config:"replacement" validate:"required"`
}
```

Reply:
Ah, didn't even occur to me.